src/main/kotlin/uk/gov/dluhc/printapi/service/PrintRequestsFileProducer.kt

```diff
@@ -111,19 +111,30 @@ class PrintRequestsFileProducer {
                 eroDeliveryTownEn ?: "",
                 eroDeliveryAreaEn ?: "",
                 eroDeliveryPostcodeEn,
-                eroNameCy ?: "",
-                eroPhoneNumberCy ?: "",
-                eroEmailAddressCy ?: "",
-                eroWebsiteCy ?: "",
-                eroDeliveryStreetCy ?: "",
-                eroDeliveryPropertyCy ?: "",
-                eroDeliveryLocalityCy ?: "",
-                eroDeliveryTownCy ?: "",
-                eroDeliveryAreaCy ?: "",
-                eroDeliveryPostcodeCy ?: "",
+                getWelshValue(eroNameCy, eroNameEn),
+                getWelshValue(eroPhoneNumberCy, eroPhoneNumberEn),
+                getWelshValue(eroEmailAddressCy, eroEmailAddressEn),
+                getWelshValue(eroWebsiteCy, eroWebsiteEn),
+                getWelshValue(eroDeliveryStreetCy, eroDeliveryStreetEn),
+                getWelshValue(eroDeliveryPropertyCy, eroDeliveryPropertyEn),
+                getWelshValue(eroDeliveryLocalityCy, eroDeliveryLocalityEn),
+                getWelshValue(eroDeliveryTownCy, eroDeliveryTownEn),
+                getWelshValue(eroDeliveryAreaCy, eroDeliveryAreaEn),
+                getWelshValue(eroDeliveryPostcodeCy, eroDeliveryPostcodeEn)
             )
```

Comment:
This looks wrong as it appears that we could get a mix of Welsh ERO sourced data and English ERO sourced data - e.g. street populated from `eroDeliveryStreetCy` but property from `eroDeliveryPropertyEn`
I guess in practice if the data is missing from the Welsh translation, it is likely to be missing from the English dataset too.  But would suggest it might be better if a single decision is made to use the Welsh ERO data or the English ERO data.  Maybe the `PrintRequestMapper` could populate the `PrintRequest` with the appropriate English or Welsh data?

Reply:
Thanks, @Neil-Massey , for your comment.
I tried to explain the implementation in the above comment, where I prevent mixing of the English and Welsh values by checking if the Welsh values are missing if `eroNameCy == null`. 

There is also a test covering that scenario as well: `should write file to stream given print request given full English properties and partially missing Welsh properties and certificate language is Welsh`

In that test we expect Welsh values, where some are null being set to "" instead of filled with corresponding English values which are non-null.

Reply:
We wanted to implement the logic in this mapper, to only affect what is going to be written in the psv file, but retaining all the remaining implementation the same as before.

Reply:
Should the database store the `eroWelsh` details that are included in the `PrintRequest`?  (I mean indicate what is sent to the printer?)

Reply:
@NathanRussellValtech and I discussed this before starting the ticket and decided not to store English ERO details as Welsh details in the database in case Welsh details are missing for that Ero.

Reply:
OK - thanks for the clarification @adacayi - when I did the original implementation I expected the PSV writer to simply write out what is stored in the database so was a little surprised at the code 😄 .